src/example-types/facet.js

```diff
@@ -134,5 +138,73 @@ module.exports = {
           }),
         options
       ),
-    })),
+    }))
+
+    let missedValues = _.difference(
+      values,
+      _.map(
+        ({ name }) => (node.isMongoId ? _.toString(name) : name),
+        results.options
+      )
+    )
+
+    let getMissedValues = (node, missedValues) =>
+      node.isMongoId ? _.map(ObjectID, missedValues) : missedValues
+
+    if (!_.isEmpty(missedValues)) {
+      let MissedResult = await search(
+        _.compact([
+          {
+            $match: {
+              [node.field]: { $in: getMissedValues(node, missedValues) },
+            },
+          },
+          { $group: { _id: `$${node.field}`, count: { $sum: 1 } } },
+          ...sortAndLimitIfNotSearching(node.optionsFilter, node.size),
+          ...lookupLabel(node),
+          _.get('label.fields', node) && projectStageFromLabelFields(node),
+          mapKeywordFilters(node),
+        ])
+      )
+      // when the value has been selected but filtered by query,the search result will still not contain the checked value. we return it with count 0
+      let missedByFilterValues = _.difference(
+        //when values are numeric values, stringify missedValues to avoid the bug.
+        _.map(_.toString, missedValues),
+        _.map(x => _.toString(x[`${node.field}`]), MissedResult)
+      )
+      let missedByFilterResult = []
+
+      if (!_.isEmpty(missedByFilterValues)) {
+        //use config to run runSearch(options, node, schema, filters, aggs)  function
+        missedByFilterResult = await config.getProvider(node).runSearch(
+          config.options,
+          node,
+          config.getSchema(node.schema),
```

Comment:
I think it's worth factoring out a util like this at the top level:

```js
let runSearch = ({ options, getSchema, getProvider }, node) => (filters, aggs) =>
  getProvider(node).runSearch(options, node, getSchema(node.schema), filters, aggs)
```

to clearly separate out this bit of contexture trivia from the details of `facet`

Reply:
long term, we'd add a curried search without filters or something to the core